firmware/src/app/model.cpp

```
@@ -29,4 +36,35 @@ bool is_empty(const engine::State& state) {
+bool is_empty(const engine::Song& song) {
+  if (song.arrangement_length > 0) return false;
+  for (const engine::State& section : song.sections) {
+    if (!is_empty(section)) return false;
+  }
+  return true;
+}
+
+// Mute is transient — a pad is muted while it is held (§6.2) — and no code carries
+// it, so the copy drops it: two songs that differ only in a held pad are the same
+// song, and holding one writes no card.
+void song_of(const Model& model, engine::Song& song) {
```

Cppcheck (2.21.0)

[style] The function 'song_of' is never used.

(unusedFunction)